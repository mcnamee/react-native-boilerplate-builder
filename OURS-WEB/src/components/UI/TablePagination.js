/* global window */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Pagination, PaginationItem } from 'reactstrap';

const Component = ({
  pagination, length, total, loading,
}) => (
  <React.Fragment>
    {(pagination && pagination.length > 0 && !loading) && (
      <div className="d-flex justify-content-between w-100">
        <p className="text-muted font-italic mt-2">{`Showing ${length} of ${total}`}</p>

        <Pagination aria-label="Pagination">
          {pagination.map((page) => (
            <PaginationItem
              key={page.link}
              disabled={(window.location.pathname === page.link)}
            >
              <Link to={page.link} className="page-link">{page.title}</Link>
            </PaginationItem>
          ))}
        </Pagination>
      </div>
    )}
  </React.Fragment>
);

Component.propTypes = {
  pagination: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    link: PropTypes.string.isRequired,
  })),
  length: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  total: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  loading: PropTypes.bool,
};

Component.defaultProps = {
  pagination: [],
  length: 0,
  total: 0,
  loading: false,
};

export default Component;
