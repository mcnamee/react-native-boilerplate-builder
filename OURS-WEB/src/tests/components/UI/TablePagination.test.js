import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import TablePagination from '../../../components/UI/TablePagination';

it('<TablePagination /> renders correctly', () => {
  const Component = (
    <MemoryRouter>
      <TablePagination
        pagination={[
          {
            title: '1',
            link: '/articles/',
          }, {
            title: '2',
            link: '/articles/2',
          },
        ]}
        length={6}
        total={12}
        loading={false}
      />
    </MemoryRouter>
  );

  // Matches snapshot
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();

  // Has the correct text on the page
  const { getByText } = render(Component);
  expect(getByText('Showing 6 of 12')).toBeInTheDocument();
  expect(getByText('2')).toBeInTheDocument();
});

it('<TablePagination /> renders correctly when no pagination set', () => {
  const Component = (
    <MemoryRouter>
      <TablePagination
        pagination={[]}
        length={6}
        total={12}
        loading={false}
      />
    </MemoryRouter>
  );

  // Matches snapshot
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toEqual(null);
});

it('<TablePagination /> renders correctly when loading', () => {
  const Component = (
    <MemoryRouter>
      <TablePagination
        pagination={[
          {
            title: '1',
            link: '/articles/',
          }, {
            title: '2',
            link: '/articles/2',
          },
        ]}
        length={6}
        total={12}
        loading
      />
    </MemoryRouter>
  );

  // Matches snapshot
  const tree = renderer.create(Component).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toEqual(null);
});
