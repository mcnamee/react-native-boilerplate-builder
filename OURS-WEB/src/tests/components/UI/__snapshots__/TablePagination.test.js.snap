// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TablePagination /> renders correctly 1`] = `
<div
  className="d-flex justify-content-between w-100"
>
  <p
    className="text-muted font-italic mt-2"
  >
    Showing 6 of 12
  </p>
  <nav
    aria-label="Pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <a
          className="page-link"
          href="/articles/"
          onClick={[Function]}
        >
          1
        </a>
      </li>
      <li
        className="page-item"
      >
        <a
          className="page-link"
          href="/articles/2"
          onClick={[Function]}
        >
          2
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<TablePagination /> renders correctly when loading 1`] = `null`;

exports[`<TablePagination /> renders correctly when no pagination set 1`] = `null`;
